Include column comments in generated types

danvk opened the pull request:
Postgres lets you attach comments to columns via the `COMMENT` command: https://www.postgresql.org/docs/current/sql-comment.html

This PR includes those as JSDoc / TSDoc comments in the types files that PgTyped generates. This has the advantage of making the comments appear in contextually appropriate places in editors that work with the TS language service, e.g. VS Code:

<img width="647" alt="image" src="https://user-images.githubusercontent.com/98301/157541840-4ab31b63-7f86-4bdf-b1c9-234af5ae7e69.png">

This seems like an overall usability win.

(Broader context: I mostly care about Postgres comments because I use them to attach TS Types to `json` / `jsonb` columns in Postgres, see https://github.com/danvk/pg-to-ts#json-types. So having these comments in source is pretty important for me. I think PgTyped is a great tool and I've got a few other contributions I'd like to make — would love to discuss them more in whatever what is most convenient.)

bradleyayers commented on packages/cli/src/generator.ts:
```diff
@@ -31,7 +32,11 @@ export const generateInterface = (interfaceName: string, fields: IField[]) => {
-    .map(({ fieldName, fieldType }) => `  ${fieldName}: ${fieldType};`)
+    .map(
+      ({ fieldName, fieldType, comment }) =>
+        (comment ? `  /** ${comment} */\n` : '') +
```
`comment` should be comment encoded to handle the case where it includes `*/`. I'm not sure how you want to do that, perhaps just strip out any `*/` from it?

danvk replied:
Good catch -- it looks like the right way to do this is to escape the trailing `/`:

```ts
/** this is a /* comment *\/ in a comment */
```

bradleyayers commented:
This is a pretty neat idea, I like it!

danvk commented:
> This is a pretty neat idea, I like it!

Thanks!

> > I use them to attach TS Types to json / jsonb columns in Postgres
> 
> Interesting approach. Bit of a nitpick, but: in my mind this type of mapping is probably better off existing in the configuration of the CLI (e.g. pg-to-ts, or pgtyped), rather than in the PG schema itself. This would be useful in a scenario where you have multiple languages (e.g. TS + Java) interfacing with the same database. Either that or use some namespacing approach like what graphile does with smart tags (e.g. `@tstype MyInterface`).

On some level the types should be included in the DB schema -- but unless you're going to use a composite type (which I believe are considered [poor form](https://github.com/brianc/node-postgres/issues/1469#issuecomment-335325883)) I don't really see how to do that. Even for a multi-language scenario, I think it's still helpful to know the expected type of the JSON column. Whatever CLI you use to generate the Java code could either match or ignore the `@type`, too. I could see an argument for `@tstype` if you expected different naming conventions in the different languages.

adelsz reviewed (approved):
Really like the idea! Don't see any immediate problems with the implementation either.
Looks good to me, thanks @danvk!

adelsz commented:
> (Broader context: I mostly care about Postgres comments because I use them to attach TS Types to `json` / `jsonb` columns in Postgres, see https://github.com/danvk/pg-to-ts#json-types. So having these comments in source is pretty important for me. I think PgTyped is a great tool and I've got a few other contributions I'd like to make — would love to discuss them more in whatever what is most convenient.)

Agree with @bradleyayers that this might be problematic in some cases because it manually couples the DB schema to specific TS types.
It does inspire some interesting ideas though, for example it might make sense to include a JSON schema in such comments and then generate a type from that schema in pgtyped 🙂

adelsz commented:
One minor request @danvk, I would really appreciate if you could add a mention of this new feature to the docs. Thanks!

danvk commented:
Sure thing, I updated the example in `ts-file-intro.md` to select the `age` column, so that you can see behavior, and I added a note about it.

Thanks for the review @adelsz -- I saw the location listed in your GitHub profile and really hope you're safe and doing OK.

danvk commented:
@adelsz this should be ready for another look

adelsz reviewed (approved):
Thanks again @danvk. Great PR
